services/bls_aggregation/blsagg.go

```diff
@@ -370,11 +370,26 @@ func checkIfStakeThresholdsMet(
 	quorumThresholdPercentagesMap map[types.QuorumNum]types.QuorumThresholdPercentage,
 ) bool {
 	for quorumNum, quorumThresholdPercentage := range quorumThresholdPercentagesMap {
+		signedStakeByQuorum, ok := signedStakePerQuorum[quorumNum]
+		if !ok {
+			// signedStakePerQuorum not contain the quorum,
+			// this case means signedStakePerQuorum has not signed for each quorum.
+			// even the total stake for this quorum is zero.
+			return false
+		}
+
+		totalStakeByQuorum, ok := totalStakePerQuorum[quorumNum]
+		if !ok {
+			// Note this case should not happend, if not found in totalStakePerQuorum
+			// We can just think is zero
```

Comment:
What do you mean by "if not found in totalStakePerQuorum"?

Actually I think this edge case could cause a bug? What if we for some reason forget to initialize totalStakeByQuorum before passing it in, but we've already seen a signer? Then we'll get signedStakeByQuorum > 0 so we'll return, even though this might just be a bug in our totalStakeByQuorum initialization.

Think we should instead log an error here "this should never happen! code must be buggy!" and return false?

Reply:
yes, i think it shoud return false with a error log